main.go

```diff
@@ -0,0 +1,208 @@
+)
+
+// Configs ...
+type Configs struct {
+	ProjectLocation string `env:"project_location,required"`
+	APKPathPattern  string `env:"apk_path_pattern"`
+	Variant         string `env:"variant"`
+	Module          string `env:"module"`
+}
+
+func failf(f string, args ...interface{}) {
+	log.Errorf(f, args...)
+	os.Exit(1)
+}
+
+func main() {
+	var config Configs
+
+	if err := stepconf.Parse(&config); err != nil {
+		failf("Couldn't create step config: %v\n", err)
+	}
+
+	stepconf.Print(config)
+
+	deployDir := os.Getenv("BITRISE_DEPLOY_DIR")
+
+	log.Printf("- Deploy dir: %s", deployDir)
+	fmt.Println()
+
+	gradleProject, err := gradle.NewProject(config.ProjectLocation)
+	if err != nil {
+		failf("Failed to open project, error: %s", err)
+	}
+
+	buildTask := gradleProject.
+		GetModule(config.Module).
+		GetTask("assemble")
+
+	log.Infof("Variants:")
+	fmt.Println()
+
+	variants, err := buildTask.GetVariants()
+	if err != nil {
+		failf("Failed to fetch variants, error: %s", err)
+	}
+
+	filteredVariants := variants.Filter(config.Variant)
+
+	for _, variant := range variants {
+		if sliceutil.IsStringInSlice(variant, filteredVariants) {
+			log.Donef("✓ %s", variant)
+		} else {
+			log.Printf("- %s", variant)
+		}
+	}
+
+	fmt.Println()
+
+	if len(filteredVariants) == 0 {
+		errMsg := fmt.Sprintf("No variant matching for: (%s)", config.Variant)
+		if config.Module != "" {
+			errMsg += fmt.Sprintf(" in module: [%s]", config.Module)
+		}
+		failf(errMsg)
+	}
+
+	if config.Variant == "" {
+		log.Warnf("No variant specified, build will run on all variants")
+		fmt.Println()
+	}
+
+	started := time.Now()
+
+	log.Infof("Run build:")
+	if err := buildTask.Run(filteredVariants); err != nil {
+		failf("Build task failed, error: %v", err)
+	}
+	fmt.Println()
+
+	log.Infof("Export APKs:")
+	fmt.Println()
+
+	apks, err := gradleProject.FindArtifacts(started, config.APKPathPattern, false)
+	if err != nil {
+		failf("failed to find apks, error: %v", err)
+	}
+
+	if len(apks) == 0 {
+		log.Warnf("No apks found with pattern: %s", config.APKPathPattern)
+		log.Warnf("If you have changed default APK export path in your gradle files then you might need to change APKPathPattern accordingly.")
+		os.Exit(0)
+	}
+
+	var exportedArtifactPaths []string
+
+	for _, apk := range apks {
+		exists, err := pathutil.IsPathExists(filepath.Join(deployDir, apk.Name))
+		if err != nil {
+			failf("failed to check path, error: %v", err)
+		}
+
+		artifactName := filepath.Base(apk.Path)
+
+		if exists {
+			timestamp := time.Now().Format("20060102150405")
+			ext := filepath.Ext(apk.Name)
+			name := strings.TrimSuffix(filepath.Base(apk.Name), ext)
+			apk.Name = fmt.Sprintf("%s-%s%s", name, timestamp, ext)
+		}
+
+		log.Printf("  Export [ %s => $BITRISE_DEPLOY_DIR/%s ]", artifactName, apk.Name)
+
+		if err := apk.Export(deployDir); err != nil {
+			log.Warnf("failed to export apks, error: %v", err)
+		}
+
+		exportedPath := filepath.Join(deployDir, apk.Name)
```

Comment:
The new variable is unnecessary: `exportedArtifactPaths = append(exportedArtifactPaths, filepath.Join(deployDir, apk.Name)`